### ISSUE TITLE
WARNING [ 'gpu_metrics file does not exist!' ]

### ISSUE THREAD

**Issue by kaijieshi7** [microsoft/nni#2587]
**Environment**:ubuntu 16.04
- NNI version:1.6
- NNI mode (local|remote|pai):local
- Client OS:ubuntu
- Server OS (for remote mode only):
- Python version:3.7
- PyTorch/TensorFlow version:torch 1.3
- Is conda/virtualenv/venv used?:yes
- Is running in Docker?:no

**Log message**:
 - nnimanager.log: 
![gpu_metrics](https://user-images.githubusercontent.com/19171201/85299528-f505fc80-b4d7-11ea-8ac0-3468da4a5c2d.png)
![dispatcher](https://user-images.githubusercontent.com/19171201/85299579-03ecaf00-b4d8-11ea-91ad-64562ded8aaf.png)
![nvidia-smi](https://user-images.githubusercontent.com/19171201/85299592-07803600-b4d8-11ea-8e3e-b11a95580552.png)

 - dispatcher.log:
 - nnictl stdout and stderr:
 

**What issue meet, what's expected?**:

**How to reproduce it?**: 

**Additional information**:

**Comment by SparkSnail** [comment]
Hi @kaijieshi7, could you please check if there is metric file under `/tmp/{.username}/nni/script`? and use `ps -ef | grep nni` to check if there is a `nni_gpu_metric` backend process? kill all of backend process related of nni, and restart the experiment.

**Comment by kaijieshi7** [comment]
where is '/tmp/{.username}/nni/script'，i cant find it

**Comment by SparkSnail** [comment]
> where is '/tmp/{.username}/nni/script'，i cant find it

{username} means your username in your system, for example, `/tmp/matrix/nni/script`

**Comment by kaijieshi7** [comment]
Do i need run some code?
Im not run some code relate with 'nni', and i cant find '/tmp/matrix/nni/script'

**Comment by SparkSnail** [comment]
Hi @kaijieshi7 , when NNI start an experiment, it will write gpu metric file to `/tmp/matrix/nni/scripts/gpu_metrics` file. Since your log has a `gpu metric file does no exist` error, I wonder if this file is created correctly. 
You can use 
```
METRIC_OUTPUT_DIR=./ python3 -m nni_gpu_tool.gpu_metrics_collector
```
script to test generating gpu_metrics file in you machine.

**Comment by kaijieshi7** [comment]
i run  'METRIC_OUTPUT_DIR=./ python3 -m nni_gpu_tool.gpu_metrics_collector',
![Screenshot from 2020-06-26 13-54-32](https://user-images.githubusercontent.com/19171201/85825385-ab0d6700-b7b4-11ea-8751-f72ec4b0bee6.png)
and when i run other's code, i can find '/tmp/matrix/nni/scripts/gpu_metrics' file ;but the gpu is   still can't work, just like what i have post first.

**Comment by SparkSnail** [comment]
Hi @kaijieshi7 , I noticed that in your log file, you required 2 gpus per trial, but seems your machine only have 1 gpu? And you should notice that NNI will not use gpus which have another process running on it,  you could set `useActiveGPU` configuration to use these busy gpus, and set `maxTrialNumPerGpu` to specify how many trial jobs on one gpu, refer https://github.com/microsoft/nni/blob/master/docs/en_US/Tutorial/ExperimentConfig.md#useactivegpu.

**Comment by kaijieshi7** [comment]
666,it's  useful , thank you guys.

**Comment by 15501388056** [comment]
My environment is WINDOWS10.I got similar problem.When I set gpuNum:0 ,it turned out FAIL.Or I set gpuNum:1,it turned out WAITING,and WARNING [ 'gpu_metrics file does not exist!' ] .How could solve this problem on WINDOWS10.

**Comment by Davidxswang** [comment]
I ran into the same problem in Windows 10. But I fixed it by 
1. running the command:  $env:METRIC_OUTPUT_DIR = 'd:/'
2. copy the generated gpu_metrics file into the temp folder

How to reproduce it?
Open nni/examples/trials/mnist-pytorch/config_windows.yml, change the gpuNum into 1, add localConfig: useActiveGpu: true, then run the experiment.
![image](https://user-images.githubusercontent.com/31802298/113290503-55a64d80-9324-11eb-824d-fb878995c82a.png)

Hopefully my solution can help some of the people who may run into the same problem in the future.

**Comment by YCccc-git** [comment]
> I ran into the same problem in Windows 10. But I fixed it by
> 
> 1. running the command:  $env:METRIC_OUTPUT_DIR = 'd:/'
> 2. copy the generated gpu_metrics file into the temp folder
> 
> How to reproduce it? Open nni/examples/trials/mnist-pytorch/config_windows.yml, change the gpuNum into 1, add localConfig: useActiveGpu: true, then run the experiment. ![image](https://user-images.githubusercontent.com/31802298/113290503-55a64d80-9324-11eb-824d-fb878995c82a.png)
> 
> Hopefully my solution can help some of the people who may run into the same problem in the future.



> 1. $env:METRIC_OUTPUT_DIR = 'd:/'

I am glad to see your suggestion, but please tell me how the command " $env:METRIC_OUTPUT_DIR = 'd:/'" works under win

**Comment by coco11563** [comment]
Hi, all fellows. I ran into this warning recently, my solution is quite easy. (On ubuntu)
step1: export the env
`export METRIC_OUTPUT_DIR="/home/username/tmp"`
step2: create the file by running code:
`python3 -m nni_gpu_tool.gpu_metrics_collector`
after the json output like:
`{"Timestamp": "Thu Sep  8 13:34:10 2022", "gpuCount": 8, "gpuInfos": [{"activeProcessNum": 12, "gpuMemUtil": "0", "gpuUtil": "0", "index": 0}, {"activeProcessNum": 10, "gpuMemUtil": "0", "gpuUtil": "0", "index": 1}, {"activeProcessNum": 1, "gpuMemUtil": "0", "gpuUtil": "0", "index": 2}, {"activeProcessNum": 1, "gpuMemUtil": "0", "gpuUtil": "0", "index": 3}, {"activeProcessNum": 1, "gpuMemUtil": "0", "gpuUtil": "0", "index": 4}, {"activeProcessNum": 1, "gpuMemUtil": "0", "gpuUtil": "0", "index": 5}, {"activeProcessNum": 1, "gpuMemUtil": "0", "gpuUtil": "0", "index": 6}, {"activeProcessNum": 1, "gpuMemUtil": "0", "gpuUtil": "0", "index": 7}]}`
then run the
`nnictl --create xxx`
No more warning pop out.

**Comment by WeigangLu** [comment]
> > Hi @kaijieshi7 , when NNI start an experiment, it will write gpu metric file to `/tmp/matrix/nni/scripts/gpu_metrics` file. Since your log has a `gpu metric file does no exist` error, I wonder if this file is created correctly. You can use
> > ```
> > METRIC_OUTPUT_DIR=./ python3 -m nni_gpu_tool.gpu_metrics_collector
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > script to test generating gpu_metrics file in you machine.
> 
> I met the same question, "gpu_metrics file does not exist!"
> 
> and while run the following code directly: METRIC_OUTPUT_DIR=./ python3 -m nni_gpu_tool.gpu_metrics_collector
> 
> it replys 'nni_gpu_tool' not found, so after 'ps -ef | grep nni', I found it changed to nni.tools.gpu_tool.gpu_metrics_collector
> 
> so I changed it to the following code, and run again: METRIC_OUTPUT_DIR=./ python3 -m nni.tools.gpu_tool.gpu_metrics_collector
> 
> but met the following error, I didn't know how to do next. ![image](https://user-images.githubusercontent.com/26847296/138846467-384e8d1f-9466-40bd-bcca-dbfb6c582c2b.png)

I also have this issue. How do you fix it?

**Comment by redLinmumu** [comment]
> Hi @kaijieshi7, could you please check if there is metric file under `/tmp/{.username}/nni/script`? and use `ps -ef | grep nni` to check if there is a `nni_gpu_metric` backend process? kill all of backend process related of nni, and restart the experiment.

I have a pid which is running, but I can not kill it because it belongs to other person. What else can I do?

avahi       1880       1  0 7月09 ?       00:06:16 avahi-daemon: running [inspur-NF5468M6-2.local]
